From cef3d991c5c19933f383706f812f29bb89f393eb Mon Sep 17 00:00:00 2001
From: Michael Haubenwallner <michael.haubenwallner@ssi-schaefer.com>
Date: Tue, 21 May 2019 18:32:09 +0200
Subject: [PATCH 11/19] run AC_HEADER_STDC check early

The first header-check resolves AC_REQUIRE(AC_PROG_EGREP). If that is
not run due to some condition, EGREP is unset for subsequent checks,
breaking the checks for function prototypes.
---
 configure.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 6cdadb4..849ea35 100644
--- a/configure.in
+++ b/configure.in
@@ -411,6 +411,8 @@ ac_link="$ac_link "'$EXTRA_LIBS'
 AC_SUBST(EXTRA_CXXFLAGS)
 ac_compile="$ac_compile "'$EXTRA_CXXFLAGS'
 
+AC_HEADER_STDC
+
 #
 # thread support
 #
@@ -1385,7 +1387,6 @@ CURSES_LIBS="$LIBS $SHLIBS"
 # Checks for header files.
 #
 
-AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h unistd.h sys/select.h strings.h float.h ieeefp.h)
 AC_CHECK_HEADERS(sys/un.h netinet/in.h arpa/inet.h netdb.h dlfcn.h dl.h)
 AC_CHECK_HEADERS(netinet/tcp.h stdlib.h sys/time.h sys/timeb.h sunmath.h sys/stat.h)
-- 
2.19.2

